src/main/java/org/opentripplanner/model/RealtimeUpdate.java

```diff
@@ -0,0 +1,33 @@
+package org.opentripplanner.model;
+
+import java.time.LocalDate;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import org.opentripplanner.transit.model.network.TripPattern;
+import org.opentripplanner.transit.model.timetable.TripOnServiceDate;
+import org.opentripplanner.transit.model.timetable.TripTimes;
+
+/**
+ * Represents the real-time update of a single trip.
+ * @param pattern the pattern to which belongs the updated trip. This can be a new pattern created in real-time.
+ * @param updatedTripTimes the new trip times for the updated trip.
+ * @param serviceDate the service date for which this update applies (updates are valid only for one service date)
+ * @param addedTripOnServiceDate optionally if this trip update adds a new trip, the TripOnServiceDate corresponding to this new trip.
+ * @param isAddedTrip true if this update creates a new trip, not present in scheduled data.
+ * @param isAddedRoute true if an added trip cannot be registered under an existing route and a new route must be created.
+ */
+public record RealtimeUpdate(
+  @Nonnull TripPattern pattern,
+  @Nonnull TripTimes updatedTripTimes,
+  @Nonnull LocalDate serviceDate,
+  @Nullable TripOnServiceDate addedTripOnServiceDate,
+  boolean isAddedTrip,
+  boolean isAddedRoute
```

Comment:
The idea was to be able to distinguish boolean and non-boolean fields in the record. Otherwise you get `addedRoute()` that returns a `boolean` and `addedTripOnServiceDate()` that returns a `TripOnServiceDate`, which is confusing.
Instead I renamed the boolean fields and accessors to to `routeCreation()`, `tripCreation()`, `tripPatternCreation()`

Reply:
Ok, conflicting Java best practices - love it :-) You may keep it. Anyone seen any best practice on this?

Reply:
I think using the type system to deal with the various cases is a nice solution to this. Sort of like a the command pattern.

Reply:
But that is well beyond the scope of this PR.